fixedwidth/src/main/java/org/apache/metamodel/fixedwidth/FixedWidthReader.java

```diff
@@ -43,6 +47,7 @@
     private final boolean _constantWidth;
     private volatile int _rowNumber;
     protected final BufferedInputStream _stream;
+    protected Reader _reader;
```

Comment:
I would suggest to keep having final fields only, and since `_stream` and `_charset` are no longer really used, I'd suggest to remove those and instead make the `initReader` return the created Reader with the needed arguments, like this:

```
_reader = initReader(stream, charsetName);
```